tests/unittestcoverage.py

```diff
@@ -194,25 +194,21 @@ def check_component(path):
   out = os.popen("gcov -p -r -s " + path + " `find " + buildpath +
                  " -name *.gcno`").read()
   dprint(out)
-  endpoint = len(out) - 1
-  while (out[endpoint] == '\n'):
-    endpoint = endpoint - 1
-  startpoint = endpoint
-  endpoint = endpoint + 1
-  while (out[startpoint] != '\n' and startpoint >= 0):
-    startpoint = startpoint - 1
-  startpoint = startpoint + 1
-
-  lastline = out[startpoint:endpoint]
-  m = re.match("Lines executed:(\d+.\d+)% of (\d+)$", lastline)
-  if m:
-    rate = float(m.group(1))
-    lines = int(m.group(2))
-  else:
-    print("ERROR! Cannot parse gcov result!")
-    return (-1, -1)
-
-  return (lines, rate)
+
+  total_lines = 0
+  total_covered = 0
+  # Calculate a line coverage per file
+  for each_line in out.splitlines():
+    m = re.match("Lines executed:(\d+.\d+)% of (\d+)$", each_line)
+    if m:
+      rate = float(m.group(1))
+      lines = int(m.group(2))
+
+      total_lines = total_lines + lines
+      total_covered = total_covered + (rate * lines)
+  total_rate = total_covered / total_lines
```

Comment:
you can add a check here that total_lines is not 0.

Reply:
Updated